### PR TITLE
Webinars page

### PR DESCRIPTION
**Pull request by RoyEJohnson**
For: https://github.com/openstax/openstax.org/issues/2274

### CODE REVIEW

**Review by jivey — CHANGES_REQUESTED** [review]
Still working my way through but wanted to get some feedback going

File: src/app/components/explore-by-collection/explore-by-collection.scss [hunk 1]
```diff
@@ -1,14 +1,10 @@
...
-    width: 100%;
-
     .cards {
         display: grid;
         grid-template-columns: repeat(auto-fill, minmax(25rem, 1fr));
         gap: 3rem;
-        margin-top: 2rem;
         width: 100%;
 
         .card {
```
[the remark this answers is not in the document]
Comment (RoyEJohnson) [reply src/app/components/explore-by-collection/explore-by-collection.scss:10]: The icons are likely to match better than these examples, but the labels will probably vary between 1 and 2 lines. I've put a min-height on the image container so size differences won't be so dramatic.
image

File: src/app/components/paginator/simple-paginator.js [hunk 2]
```diff
@@ -96,3 +96,21 @@ export default function SimplePaginator({currentPage, setPage, totalPages}) {
...
+    return [start, end];
+}
+
+export function Showing({page, perPage=9, totalCount, ofWhat}) {
+    const [start, end] = itemRangeOnPage(page, perPage, totalCount);
+    const countMessage = totalCount <= perPage ? 'all' : `${start}-${end} of`;
+
+    return (
+        <div className="whats-showing">
+            Showing {countMessage} {totalCount} {ofWhat}
```
[the remark this answers is not in the document]
Comment (RoyEJohnson) [reply src/app/components/paginator/simple-paginator.js:113]: This is desired behavior. It's letting them know how many matches and that they are all displayed (as opposed to being paginated).

File: src/app/helpers/use-data.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+    if (processing.transform) {
+        // mutates
+        transformData(rawData);
+    }
+    if (processing.camelCase) {
+        returnValue = camelCaseKeys(rawData);
+    }
+    if (processing.postProcess) {
+        returnValue = (returnValue as Array<unknown>).map(processing.postProcess);
```
[the remark this answers is not in the document]
Comment (RoyEJohnson) [reply src/app/helpers/use-data.ts:84]: The postProcess step will necessarily be an array operation. The CMS can send other data, but you wouldn't use `postProcess` on it.

File: src/app/helpers/use-data.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    const promise: Promise<Response> = React.useMemo(
+        () => slugPromise ?? urlPromise ?? promiseOption,
+        [slugPromise, urlPromise, promiseOption]
+    );
+    const processedPromise = React.useMemo(
+        () =>
+            promise
+                .then((resp) => resp[options.resolveTo]())
+                .then((rawData) => processRawData<T>(rawData, options)),
+        [promise, options.resolveTo] // eslint-disable-line react-hooks/exhaustive-deps
```
[the remark this answers is not in the document]
Comment (RoyEJohnson) [reply src/app/helpers/use-data.ts:56]: `options`. To be excruciatingly correct, I'd need to unpack the components of `options` into separate memo'd items and then package them up into another memo.

**Review by jivey — CHANGES_REQUESTED** [review]
Overall looking good, just a couple issues I noticed with the JS date and alt text

File: src/app/pages/webinars/webinar-cards/latest-webinars.tsx [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        hour: 'numeric',
+        minute: 'numeric',
+        timeZoneName: 'short'
+    });
+
+    return (
+        <div className='card upcoming'>
+            <div className='dated-heading'>
+                <div className='date'>
+                    <div className='day-of-month'>{data.start.getDay()}</div>
```
[the remark this answers is not in the document]
Comment (RoyEJohnson) [reply src/app/pages/webinars/webinar-cards/latest-webinars.tsx:86]: D'oh!

File: src/app/pages/webinars/explore-page/explore-page.tsx [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+                if (exploreType === 'collections') {
+                    const associations = w.collections;
+                    const hasAssociation = associations.some(
+                        (a) =>
+                            a.collection === topic &&
+                            (!sectionKey || a[sectionKey] === 'True')
+                    );
+
+                    return hasAssociation;
+                }
```
[the remark this answers is not in the document]
Comment (RoyEJohnson) [reply src/app/pages/webinars/explore-page/explore-page.tsx:88]: I initially tried to write it that way, but I got stymied by TypeScript to the point that I figured it wanted me to separate them out.

**Review by jivey — APPROVED** [review]

**Pull request merged**